### ISSUE TITLE
0.28.7: hang when sent `initialize` message

### ISSUE THREAD

**Issue by siegel** [pappasam/jedi-language-server#105]
Following on from #104,  it looks as though there may be a different bug that is causing the server to hang at startup.

Repro:
```
pip uninstall jedi-language-server
pip install -U jedi-language-server
```

Start the server, and send it an `initialize` message with these parameters (sorry, this is the output of printing an `NSDictionary` and thus is not JSON but can probably be turned into it easily enough if you can't repro in isolation):

```
{
    capabilities =     {
        completion =         {
            completionItem =             {
                deprecatedSupport = 1;
                insertReplaceSupport = 1;
                insertTextModeSupport =                 {
                    valueSet =                     (
                        1,
                        2
                    );
                };
                preselectSupport = 1;
                snippetSupport = 1;
            };
        };
        textDocument =         {
            codeAction =             {
            };
            completion =             {
                completionItem =                 {
                    deprecatedSupport = 1;
                    insertReplaceSupport = 1;
                    insertTextModeSupport =                     {
                        valueSet =                         (
                            1,
                            2
                        );
                    };
                    preselectSupport = 1;
                    resolveSupport =                     {
                        properties =                         (
                            detail,
                            documentation
                        );
                    };
                    snippetSupport = 1;
                };
            };
            documentSymbol =             {
                hierarchicalDocumentSymbolSupport = 1;
                labelSupport = 1;
            };
            publishDiagnostics =             {
                categorySupport = 1;
                codeActionsInline = 1;
                codeDescription = 1;
                dataSupport = 1;
                relatedInformation = 1;
            };
            signatureHelp =             {
                signatureInformation =                 {
                    activeParameterSupport = 1;
                    parameterInformation =                     {
                        labelOffsetSupport = 1;
                    };
                };
            };
            synchronization =             {
                didSave = 1;
                willSave = 1;
            };
        };
        workspace =         {
            workspaceFolders = 1;
        };
    };
    initializationOptions =     {
    };
    processId = 19458;
    rootUri = "<null>";
    trace = verbose;
    workspaceFolders =     (
    );
}
```

Thereupon the server never responds to the `initialize` request, and there's a Python process running consuming high CPU.

**Comment by pappasam** [comment]
@siegel has this been resolved in 0.10.3? That version is compatible with the latest release of jedi-language-server so you can test it right now!

**Comment by pappasam** [comment]
@siegel I assume this has been resolved. If it has not been, we can re-open.
